src/feature/react-hook-form/index.tsx

```diff
@@ -7,6 +7,12 @@ import { z } from 'zod';
 export const validationSchema = z.object({
   title: z.string().min(2),
   text: z.string().min(5),
+  dateStr: z
+    .string()
+    .transform((value) => new Date(value))
+    .refine((value) => !isNaN(value.getTime()), {
+      message: 'Invalid date',
+    }),
 });
```

Comment:
@bluebill1049 this validation schema is used **both** on the server and the client which means that the client will send `dateStr: Date` and not `dateStr: "..."`